src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,37 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0393: r##"
+A type parameter which references `Self` in its default value was not specified.
+Example of erroneous code:
```

Comment:
Currently it's  not a sentence, though. Neither is "Erroneous code example", but that reads more naturally so I'm mostly okay with that.